### PR TITLE
channel state sync in cluster

### PR DESCRIPTION
**Pull request by DoraALin**
1. update channel states(skip, pause) among clusters
2. skip/unskip UI in nsqadmin

pr for #30 #31 
Signed-off-by: 元守 <lulin@youzan.com>

### CODE REVIEW

File: nsqdserver/context.go [hunk 1]
```diff
@@ -175,6 +175,25 @@ func (c *context) DeleteExistingChannel(topic *nsqd.Topic, channelName string) e
...
 }
 
+func (c *context) UpdateChannelState(ch *nsqd.Channel, paused int, skipped int) error {
+	var err error
+	if c.nsqdCoord == nil {
+		switch paused {
+		case 1:
+			err = ch.Pause()
+		case 0:
+			err = ch.UnPause()
```
Comment (absolute8511) [inline nsqdserver/context.go:185]: Missing skipped here?

File: nsqdserver/http.go [hunk 2]
```diff
@@ -563,6 +565,39 @@ func (s *httpServer) doDeleteChannel(w http.ResponseWriter, req *http.Request, p
...
+	} else {
+		err = s.ctx.UpdateChannelState(channel, -1, 1)
+	}
+	if err != nil {
+		nsqd.NsqLogger().LogErrorf("failure in %s - %s", req.URL.Path, err)
+		return nil, http_api.Err{500, "INTERNAL_ERROR"}
+	}
+
+	// pro-actively persist metadata so in case of process failure
+	s.ctx.persistMetadata()
```
Comment (absolute8511) [inline nsqdserver/http.go]: Missing skipped meta data in `persistMetadata` .  The struct `ChannelMetaInfo` should be changed. And load meta from disk should be handled.

**Comment by absolute8511** [comment]
Does this including pr for #30 ? I did not see any related.

**Comment by DoraALin** [comment]
@absolute8511 add skipped info in channel meta data & test case. for your review.

File: nsqdserver/context.go (hunk 1 above)
Comment (DoraALin) [reply nsqdserver/context.go:185]: switch for skip added in https://github.com/absolute8511/nsq/pull/32/commits/3c68264fd036d0d6829f69449db871e225532122

**Comment by DoraALin** [comment]
@absolute8511  my bad, #30 is not related, it is #29 .

**Review by absolute8511 — APPROVED** [review]
LGTM

**Pull request merged**